todo-list/fn_and_obj.py

```diff
@@ -0,0 +1,129 @@
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].description == item_e:
+                return True
+        return False 
+    
+    # Finding the index of a ToDoItem allows us to then call any item method.
+    # Otherwise each ToDoItem method would require a separate ToDoList method.
+    def find_item(self, item_f):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].description == item_f:
+                return i
+        
+    # Each print option has a separate method (not condensed due to different printing)
+    def print_items(self):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].complete:
+                completed_str = ' is completed'
+            else:
+                completed_str = ' is not completed'
+            print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str + completed_str)
+            
+    def print_incomp_items(self):
+        for i in range(len(self.actual_list)):
+            if not self.actual_list[i].complete:
+                print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str)
+    
+    def print_comp_items(self):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].complete:
+                print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str)
+
+    def print_overdue_items(self):
+        for i in range(len(self.actual_list)):
+            if (not self.actual_list[i].complete) and datetime.now() > self.actual_list[i].date_obj:
+                print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str)
+
+    def print_search_items(self, search):
+        for i in range(len(self.actual_list)):
+            if search.lower() in self.actual_list[i].description.lower():
+                if self.actual_list[i].complete:
+                    completed_str = ' is completed'
+                else:
+                    completed_str = ' is not completed'
+                print(self.actual_list[i].description + ' due on ' + 
+                          self.actual_list[i].date_str + completed_str)
+
+
+# Create class for to do item
+class ToDoItem:
+    
+    def __init__(self, description, date):
+        self.description = description
+        self.complete = False # Items are always created as incomplete
+        self.date_str = date
+        self.date_obj = datetime.strptime(date, '%d/%m/%Y')
+        
+    # Date is stored as string and date format for printing and filtering
+    def edit_date(self, date):
+        self.date_str = date
+        self.date_obj = datetime.strptime(date, '%d/%m/%Y')
+        
+    def mark_as_complete(self):
+        self.complete = True
+        
+    def edit_description(self, new_description):
+        self.description = new_description
```

Comment:
One thing that's worth bearing in mind with Python is that you can always edit the attributes outside of the class.
If all the method does is to  set the value of an attribute to be what you passed in, that's not really any easier than just doing `item.description = 'foo'`, so the method feels redundant here without more complex logic.